src/gen_rules.ml

```diff
@@ -24,6 +24,62 @@ module Gen(P : Install_rules.Params) = struct
     | Some config ->
       Format_rules.gen_rules sctx config ~dir
 
+  let gen_scheme_rules ~dir =
+    let scope = SC.find_scope_by_dir sctx dir in
+    let project = Scope.project scope in
+    let add_alias ~loc ~file ~alias_name ~action_template =
+      let deps =
+        [Dune_file.Bindings.Unnamed
+           (Dune_file.Dep_conf.File
+              (String_with_vars.make_text
+                 loc
+                 (Path.to_string file)))]
+      in
+      let extra_bindings =
+        let file_path =
+          Path.relative (Scope.root scope) (Path.to_string file)
+        in
+        Pform.Map.singleton
+          "test"
+          (Pform.Var.Values [Value.Path file_path])
+      in
+      Simple_rules.alias sctx
+        ~extra_bindings
+        ~dir
+        ~scope
+        { Alias_conf.
+          name       = alias_name
+        ; deps
+        ; action     = Some (loc, action_template)
+        ; locks      = []
+        ; package    = None
+        ; enabled_if = Blang.true_
+        ; loc
+        }
+    in
+    let path_has_extension path ext =
+      Path.extension path = "." ^ ext
+    in
+    File_tree.fold (SC.file_tree sctx)
```

Comment:
Since we already detect in the file tree which directories that are part of a scheme, could we instead add `scheme` field to `File_tree.Dir`?